packages/core/src/test/testUtil.ts

```diff
@@ -121,7 +125,10 @@ export async function createTestWorkspace(
     const fileContent = opts?.fileContent ?? ''
 
     do {
-        const tempFilePath = path.join(workspace.uri.fsPath, `${fileNamePrefix}${n}`)
+        const tempFilePath = path.join(
+            workspace.uri.fsPath,
+            opts.fileNameOverride ? `${n}-${opts.fileNameOverride}` : `${n}-${fileNamePrefix}`
```

Comment:
because the current way of generating test data does not allow customizing file extensions, only file prefixes, hence this is much needed, but your point make sense, I made changes to keep the old behaviour intact when `fileName` is not provided